### ISSUE TITLE
current cross-validation looks wrong and might be unfair

### ISSUE THREAD

**Issue by siemdejong** [siemdejong/skinstression#6]
At the moment, the k-fold cross-validation algorithm is as follows;
1. loop through every fold
    1. loop through the train set
    2. loop through the validation set
    3. loop through the test set
The test and validation set are in the same loop and test on exactly the same model every instance, this doesn't make sense. Getting the test set out of the loop and testing the model only on the best model is better.

Also, does doing the following make sense? Nested cross validation:
![image](https://user-images.githubusercontent.com/28396796/193228829-0de193ab-6935-45fe-ae1b-b618680e1e3c.png)
